### PR TITLE
Enhance campaign

### PR DESCRIPTION
**Pull request by jrmi**
- Add filter for viewpoints last picture date and id
- Allow to sort by picture state

### CODE REVIEW

File: src/modules/RA/Campaign/components/AddViewpoint.js [hunk 1]
```diff
@@ -97,6 +110,22 @@ const ViewpointModal = ({ show, setShow, onAdd, ids }) => {
...
+            id="demo-simple-select-filled"
+            value={pictureDateFilter}
+            onChange={e => { setPictureDateFilter(e.target.value); }}
+            style={{ width: '13em', marginLeft: '0.5em' }}
+          >
+            <MenuItem value="">
+              <em>{translate('form.all')}</em>
+            </MenuItem>
+            {lastPictureChoices.map(
+              ({ id, name }) => <MenuItem value={id}>{translate(name)}</MenuItem>,
```
Comment (dtrucs) [inline src/modules/RA/Campaign/components/AddViewpoint.js:125]: key prop is missing

File: src/modules/RA/Picture/components/PictureFields.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ const useStyles = makeStyles({
       right: 0,
       padding: '2em',
       fontSize: '1.5em',
-      margin: '0 0 8px -8px',
+      margin: '0 -8px 0px 8px',
```
Comment (dtrucs) [inline src/modules/RA/Picture/components/PictureFields.js:59]: px of `0px` is useless

File: src/modules/RA/Viewpoint/views/ListFilters.js [hunk 3]
```diff
@@ -40,6 +40,13 @@ const reduceFilters = filterOptions => (filterList, [filterName, filterProps]) =
...
   ISODateFormat(new Date(new Date().setFullYear(new Date().getFullYear() - x)));
 
+export const lastPictureChoices = [
+  { id: moreXYears(1), name: 'resources.viewpoint.filters.more-1-year' },
+  { id: moreXYears(2), name: 'resources.viewpoint.filters.more-2-year' },
+  { id: moreXYears(3), name: 'resources.viewpoint.filters.more-3-year' },
+  { id: moreXYears(5), name: 'resources.viewpoint.filters.more-5-year' },
+];
+
 export const ListFilters = ({ ...props }) => {
```
Comment (dtrucs) [inline src/modules/RA/Viewpoint/views/ListFilters.js:50]: The spread of props seems useless here

**Review by dtrucs — CHANGES_REQUESTED** [review]

File: src/modules/RA/Campaign/components/AddViewpoint.js (hunk 1 above)
Comment (jrmi) [reply src/modules/RA/Campaign/components/AddViewpoint.js]: Done

File: src/modules/RA/Picture/components/PictureFields.js (hunk 2 above)
Comment (jrmi) [reply src/modules/RA/Picture/components/PictureFields.js]: done

File: src/modules/RA/Viewpoint/views/ListFilters.js (hunk 3 above)
Comment (jrmi) [reply src/modules/RA/Viewpoint/views/ListFilters.js]: Indeed

**Review by dtrucs — APPROVED** [review]

**Pull request merged**